flow/layers/color_filter_layer_unittests.cc

```diff
@@ -136,19 +148,28 @@ TEST_F(ColorFilterLayerTest, MultipleChildren) {
   EXPECT_EQ(mock_layer1->parent_matrix(), initial_transform);
   EXPECT_EQ(mock_layer2->parent_matrix(), initial_transform);
 
-  SkPaint filter_paint;
-  filter_paint.setColorFilter(layer_filter);
-  layer->Paint(paint_context());
-  EXPECT_EQ(
-      mock_canvas().draw_calls(),
-      std::vector({MockCanvas::DrawCall{
-                       0, MockCanvas::SaveLayerData{children_bounds,
-                                                    filter_paint, nullptr, 1}},
-                   MockCanvas::DrawCall{
-                       1, MockCanvas::DrawPathData{child_path1, child_paint1}},
-                   MockCanvas::DrawCall{
-                       1, MockCanvas::DrawPathData{child_path2, child_paint2}},
-                   MockCanvas::DrawCall{1, MockCanvas::RestoreData{0}}}));
+  DisplayListBuilder expected_builder;
+  /* ColorFilterLayer::Paint() */ {
+    expected_builder.setColorFilter(dl_color_filter.get());
+    expected_builder.saveLayer(&children_bounds, true);
+    {
+      /* MockLayer::Paint() */ {
+        expected_builder.setColor(SkColors::kYellow.toSkColor());
+        expected_builder.setColorFilter(nullptr);
+        expected_builder.drawPath(child_path1);
+      }
+      /* MockLayer::Paint() */ {
+        expected_builder.setColor(SkColors::kCyan.toSkColor());
+        expected_builder.setColorFilter(nullptr);
```

Comment:
This call to setColorFilter may have been redundant since it was already set to null in the previous block - another reason to prefer the (..., &dl_paint) variants...